### PR TITLE
[FLINK-28447] GitHub Pull Request template

### PR DESCRIPTION
**Pull request by mbalassi**
I have started from the template in the main Flink repo and made some simplification given the smaller scope of the project.

Let me know if you prefer to make the guide more concise.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<!--
+*Thank you very much for contributing to the Apache Flink Kubernetes Operator - we are happy that you want to help us improve the project. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*
+
+*Please understand that we do not do this to make contributions to the project a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*
```
Comment (gyfora) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: I think we can remove this paragraph

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+## Contribution Checklist
+
+  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
+  
+  - Name the pull request in the form "[FLINK-XXXX] [component] Title of the pull request", where *FLINK-XXXX* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
+  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix][docs] Fix typo in event time introduction` or `[hotfix][javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.
+
+  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
+  
+  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes. You can read more on how we use GitHub Actions for CI [here](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/development/guide/#cicd).
```
Comment (gyfora) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: Should we suggest the same command run by CI instead? Or does mvn verify work properly?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+*(example:)*
+  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
+  - *Extended integration test for recovery after master (JobManager) failure*
+  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*
+
+## Does this pull request potentially affect one of the following parts:
+
+  - Dependencies (does it add or upgrade a dependency): (yes / no)
+  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: (yes / no)
+
```
Comment (gyfora) [inline .github/PULL_REQUEST_TEMPLATE.md:64]: Here we could also add: Core observe/reconcile logic

**Review by gyfora — COMMENTED** [review]
Thanks @mbalassi . Looks pretty good. I added a few small comments

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+## Brief change log
+
+*(for example:)*
+  - *Periodic savepoint trigger is introduced to the custom resource*
+  - *The operator checks on reconciliation whether the required time has passed*
+  - *The JobManager's dispose savepoint API is used to clean up obsolete savepoints*
+
+## Verifying this change
+
+Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing
```
Comment (gyfora) [inline .github/PULL_REQUEST_TEMPLATE.md:41]: this should be commented out

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mbalassi) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: `mvn clean verify` works for the purposes of this.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (mbalassi) [reply .github/PULL_REQUEST_TEMPLATE.md:63]: Good call, done.

**Comment by mbalassi** [comment]
Thanks, @gyfora. Addressed your comments.

**Pull request merged**